plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -0,0 +1,364 @@
+    const providers: GitlabOrgDiscoveryEntityProvider[] = [];
+
+    providerConfigs.forEach(providerConfig => {
+      const integration = integrations.byHost(providerConfig.host);
+
+      if (!providerConfig.orgEnabled) {
+        return;
+      }
+
+      if (!integration) {
+        throw new Error(
+          `No gitlab integration found that matches host ${providerConfig.host}`,
+        );
+      }
+
+      if (!options.schedule && !providerConfig.schedule) {
+        throw new Error(
+          `No schedule provided neither via code nor config for GitlabOrgDiscoveryEntityProvider:${providerConfig.id}.`,
+        );
+      }
+
+      const taskRunner =
+        options.schedule ??
+        options.scheduler!.createScheduledTaskRunner(providerConfig.schedule!);
+
+      providers.push(
+        new GitlabOrgDiscoveryEntityProvider({
+          ...options,
+          config: providerConfig,
+          integration,
+          taskRunner,
+        }),
+      );
+    });
+    return providers;
+  }
+
+  private constructor(options: {
+    config: GitlabProviderConfig;
+    integration: GitLabIntegration;
+    logger: Logger;
+    taskRunner: TaskRunner;
+  }) {
+    this.config = options.config;
+    this.integration = options.integration;
+    this.logger = options.logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(options.taskRunner);
+  }
+
+  getProviderName(): string {
+    return `GitlabOrgDiscoveryEntityProvider:${this.config.id}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  private createScheduleFn(taskRunner: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class: GitlabOrgDiscoveryEntityProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(`${this.getProviderName()} refresh failed`, error);
+          }
+        },
+      });
+    };
+  }
+
+  private async refresh(logger: Logger): Promise<void> {
+    if (!this.connection) {
+      throw new Error(
+        `Gitlab discovery connection not initialized for ${this.getProviderName()}`,
+      );
+    }
+
+    const client = new GitLabClient({
+      config: this.integration.config,
+      logger: logger,
+    });
+
+    const users = paginated<GitLabUser>(options => client.listUsers(options), {
+      page: 1,
+      per_page: 50,
+      active: true,
+    });
+
+    const groups = paginated<GitLabGroup>(
+      options => client.listGroups(options),
+      {
+        page: 1,
+        per_page: 50,
+      },
+    );
```

Comment:
any particular reason for the 50-item pages? My gitlab instance allows up to 100
```suggestion
    const users = paginated<GitLabUser>(options => client.listUsers(options), {
      page: 1,
      per_page: 100,
      active: true,
    });

    const groups = paginated<GitLabGroup>(
      options => client.listGroups(options),
      {
        page: 1,
        per_page: 100,
      },
    );
```